pkg/storaged/nfs-mounts.py

```diff
@@ -0,0 +1,344 @@
+        if t == mount_point or t.startswith(mount_point + "/"):
+            unit = systemd_manager.GetUnitByPID(int(pid))
+            if unit not in results:
+                unit_obj = bus.get_object('org.freedesktop.systemd1', unit)
+                desc = unit_obj.Get("org.freedesktop.systemd1.Unit", "Description",
+                                    dbus_interface="org.freedesktop.DBus.Properties")
+                id = unit_obj.Get("org.freedesktop.systemd1.Unit", "Id",
+                                  dbus_interface="org.freedesktop.DBus.Properties")
+                results[unit] = { "unit": id, "desc": desc }
+            return True
+        return False
+
+    def checkdir(path, pid):
+        for f in os.listdir(path):
+            if check(os.path.join(path, f), pid):
+                return True
+        return False
+
+    my_pid = os.getpid()
+
+    for p in os.listdir("/proc/"):
+        if not p.isdigit():
+            continue
+        if int(p) == my_pid:
+            continue
+        proc = "/proc/%s/" % p
+        try:
+            if check(proc + "exe", p): continue
+            if check(proc + "root", p): continue
+            if check(proc + "cwd", p): continue
+            if checkdir(proc + "fd", p): continue
+            if checkdir(proc + "map_files", p): continue
+        except OSError:
+            pass
+
+    return list(results.values())
+
+def users(entry):
+    data = fuser(entry)
+    sys.stdout.write(json.dumps(data) + "\n")
+    sys.stdout.flush()
+
+def stop_units(units):
+    import dbus
+    bus = dbus.SystemBus()
+    systemd_manager = dbus.Interface(bus.get_object('org.freedesktop.systemd1', '/org/freedesktop/systemd1'),
+                                     dbus_interface='org.freedesktop.systemd1.Manager')
+    for u in units:
+        systemd_manager.StopUnit(u, 'replace')
+
+def stop_and_unmount(units, entry):
+    stop_units(units)
+    unmount(entry)
+
+def stop_and_remove(units, entry):
+    stop_units(units)
+    remove(entry)
+
+def dispatch(argv):
+    if argv[1] == "monitor":
+        monitor()
+    elif argv[1] == "update":
+        update(json.loads(argv[2]), json.loads(argv[3]))
+    elif argv[1] == "add":
+        add(json.loads(argv[2]))
+    elif argv[1] == "remove":
+        remove(json.loads(argv[2]))
+    elif argv[1] == "mount":
+        mount(json.loads(argv[2]))
+    elif argv[1] == "unmount":
+        unmount(json.loads(argv[2]))
+    elif argv[1] == "stop-and-unmount":
+        stop_and_unmount(json.loads(argv[2]), json.loads(argv[3]))
+    elif argv[1] == "stop-and-remove":
+        stop_and_remove(json.loads(argv[2]), json.loads(argv[3]))
+    elif argv[1] == "users":
+        users(json.loads(argv[2]))
+
+try:
+    dispatch(sys.argv)
+except subprocess.CalledProcessError as e:
+    sys.exit(e.returncode)
+except:
+    sys.stderr.write("".join(traceback.format_exception_only(sys.exc_info()[0], sys.exc_info()[1])))
+    sys.stderr.flush()
+    sys.exit(1)
```

Comment:
Is there much difference to the default exception handler? I'd rather prefer to not handle blanket unexpected exceptions, and  rather let things like Apport or ABRT kick in to catch and report these.

Reply:
Oh, I was missing the `format_exception_only()`. Seems simpler to just use `except Exception as e:` and `sys.stderr.write(e)`?

Reply:
D'oh, I and was thinking "why is it so hard to make a shorter error report"...